MiniApp/Classes/ui/MiniAppViewController.swift

```diff
@@ -224,6 +234,37 @@ public class MiniAppViewController: UIViewController {
         backButton.isEnabled = backButtonEnabled
         forwardButton.isEnabled = forwardButtonEnabled
     }
+
+    // MARK: - Sharing
+    @objc
+    public func sharePressed() {
+        MiniApp
+            .shared(with: config, navigationSettings: .none)
+            .info(miniAppId: appId, miniAppVersion: version) { result in
+                switch result {
+                case .success(let info):
+                    guard
+                        let text = info.promotionalText,
+                        let imageUrl = info.promotionalImageUrl
+                    else {
+                        return
+                    }
+                    DispatchQueue.main.async {
+                        self.showShareAlert(text: text, imageUrl: imageUrl)
+                    }
```

Comment:
Do you think we should capture self there?

```suggestion
                    DispatchQueue.main.async { [weak self] in
                        self?.showShareAlert(text: text, imageUrl: imageUrl)
                    }
```

Reply:
I think dispatch.main should not cause much issues because it should be always called but for the outer closure it's probably safer 👍